### PR TITLE
Disha/hack manifest maker

### PR DESCRIPTION
**Pull request by Dishalodha**
Changes done to remove collect files module and instead pass the grouped files to all the downstream processes

### CODE REVIEW

File: pipelines/nextflow/subworkflows/genome_prepare/main.nf [hunk 1]
```diff
@@ -98,12 +97,10 @@ workflow GENOME_PREPARE {
         prepared_files = prepared_files_grouped
                     .groupTuple()
 
-        // Collect in manifest, checks and generate sequence stats
-        collect_dir = COLLECT_FILES(prepared_files)
+        // Checks and generate sequence stats for manifest
+        manifest_bundle = MANIFEST(all_files)
```
[the remark this answers is not in the document]
Comment (Dishalodha) [reply pipelines/nextflow/subworkflows/genome_prepare/main.nf:101]: oh yes that should be '"prepared_files", thanx

File: pipelines/nextflow/modules/manifest/manifest_stats.nf [hunk 2]
```diff
@@ -18,19 +18,19 @@ process MANIFEST_STATS {
...
+        '''
         CHECK_ASSEMBLY=""
-        if [ "$ncbi_check" == 1 ]; then
-            CHECK_ASSEMBLY=" --datasets_bin $datasets --accession $accession"
+        if [ "!{ncbi_check}" == 1 ]; then
+            CHECK_ASSEMBLY=" --datasets_bin !{datasets} --accession !{meta.accession}"
         fi
-        manifest_stats --manifest_dir "$manifest_dir" \$CHECK_ASSEMBLY
-        """
+        manifest_stats --manifest_dir ./ $CHECK_ASSEMBLY
```
Comment (JAlvarezJarreta) [inline pipelines/nextflow/modules/manifest/manifest_stats.nf:34]: I would rather have this with the Nextflow nomenclature with two `'''` blocks.

File: pipelines/nextflow/subworkflows/genome_prepare/main.nf (hunk 1 above)
Comment (MatBarba) [inline pipelines/nextflow/subworkflows/genome_prepare/main.nf:101]:
```suggestion
        manifest_bundle = MANIFEST(prepared_files)
```

File: pipelines/nextflow/modules/manifest/integrity.nf [hunk 3]
```diff
@@ -15,24 +15,24 @@
...
         brc_mode=''
         if [ $brc_mode == 1 ]; then
             brc_mode='--brc_mode 1'
         fi
-        check_integrity --manifest_file ${manifest_dir}/manifest.json \
-            \$brc_mode
-        """
+        check_integrity --manifest_file ./manifest.json \
+            $brc_mode
+        '''
```
Comment (JAlvarezJarreta) [inline pipelines/nextflow/modules/manifest/integrity.nf:37]: Should we put thing as a Nextflow if-then-else block?

**Review by JAlvarezJarreta — APPROVED** [review]

File: pipelines/nextflow/modules/files/publish_output.nf [hunk 4]
```diff
@@ -14,21 +14,21 @@
...
+        path "*", includeInputs: true
     
-    script:
-        """
-        echo "Just copy over the finished files"
-        echo "From '$data_dir' to '$out_dir' for accession '$accession'"
-        """
+    shell:
+        '''
+        echo "Just publishing the finished files"
```
Comment (ens-LCampbell) [inline pipelines/nextflow/modules/files/publish_output.nf:31]: I wonder would it be better to write something more meaningful as a message....

E.g. "Gathering processed genome files (JSON, Fasta, meta) to publish to output dir !{out_dir}."

I don't mean list actual file _names_, but what kinds of files. Just a suggestion

File: pipelines/nextflow/modules/gff3/unpack_gff3.nf [hunk 5]
```diff
@@ -17,14 +17,14 @@ process UNPACK_GFF3 {
...
-        tuple val(gca), path("*.${extension}")
+        tuple val(meta), path("*.${extension}")
     
-    script:
-    """
-    extract_file --src_file ${compressed_gff} --dst_dir "."
-    """
+    shell:
+    '''
+    extract_file --src_file !{compressed_gff} --dst_dir "."
```
Comment (ens-LCampbell) [inline pipelines/nextflow/modules/gff3/unpack_gff3.nf:28]: Is it ok or at least good practice to provide current dir as --dst_dir "." ?

**Review by ens-LCampbell — APPROVED** [review]

**Pull request merged**